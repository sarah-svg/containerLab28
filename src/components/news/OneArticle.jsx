// Article - Display an articles title, author, and description
////////creates one article 

import React from 'react';
import PropTypes from 'prop-types';

const OneArticle = ({ title, author, description }) => (
  <div data-testid="display">
    <h2>{title}</h2>
    <h3>{author}</h3>
    <span>{description}</span>
  </div>
);
OneArticle.propTypes = {
  title: PropTypes.sting.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default OneArticle;
