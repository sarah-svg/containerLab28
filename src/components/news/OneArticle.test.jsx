import React from 'react';
import { render, cleanup } from '@testing-library/react';
import OneArticle from './OneArticle';

describe('Article', () => {
  afterEach(() => cleanup());
  it ('renders the article', () => {
    const { asFragment } = render(<OneArticle />);
    expect(asFragment).toMatchSnapshot();
  });
});
