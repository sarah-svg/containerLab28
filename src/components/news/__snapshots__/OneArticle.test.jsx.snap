// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders the article 1`] = `[Function]`;
